### ISSUE TITLE
 Saving box_loss, cls_loss, and dfl_loss curves

### ISSUE THREAD

**Issue by WindRise29** [ultralytics/ultralytics#17752]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How to save the box_loss, cls_loss, dfl_loss and other curves in the following figure during training and validation as images? The effect is shown in the following figure
![_20241125095559](https://github.com/user-attachments/assets/7096402c-c9ca-4fca-b755-4bc163812a09)

My training code is as follows:
```python
from ultralytics import YOLO

if __name__ == '__main__':
    # Load a model
    model = YOLO("yolo11n.pt")

    # Train the model
    model.train(
    data="D:/WorkingSpaceForCoding/yolov11/ultralytics-main/ultralytics/cfg/datasets/bone.yaml",  # path to dataset YAML
    epochs=200,  # number of training epochs
    imgsz=640,  # training image size
    device="0",  # device to run on, i.e. device=0 or device=0,1,2,3 or device=cpu
    close_mosaic=0,
    name="train"
    )
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WindRise29, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response, and an Ultralytics engineer will also assist you soon.

To help us better understand your situation, if this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). For your use case, ensure you're operating the latest version of our `ultralytics` package, as updates might resolve the issue:

```bash
pip install -U ultralytics
```

Meanwhile, if you are seeking to save loss curves such as box_loss, cls_loss, and dfl_loss, here's how you might proceed:

1. **Training Configuration**: Try expanding your current training script to include callbacks or modify events during training to save these plots.
2. **Documentation**: Check out the [Docs](https://docs.ultralytics.com) for detailed training configurations and options.
3. **Sample Dataset and Logs**: If this is a custom training ❓ Question, please provide dataset image examples and training logs. Ensure that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our thriving community to exchange ideas or seek further real-time help:
- [Discord](https://ultralytics.com/discord) 🎧 for instant discussion.
- [Discourse](https://community.ultralytics.com) for thorough conversation.
- [Subreddit](https://reddit.com/r/ultralytics) for community knowledge sharing.

## Verified Environments

Ensure a smooth experience by using our officially verified environments:
- **Notebooks** with free GPU: 
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> 
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> 
  <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Platforms**: Follow our guides for [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) and [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker**: See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/). 
  <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Current Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. Stay tuned for further updates!

**Comment by Laughing-q** [comment]
@WindRise29 The results are saved as `results.csv` in the same directory where you find the image. :)

**Comment by WindRise29** [comment]
> @WindRise29 The results are saved as `results.csv` in the same directory where you find the image. :)

These are the images I was looking for. After I finished the training, there were no curve images（I can find the CSV actually）. I found that I could save these images while reading the paper.

**Comment by glenn-jocher** [comment]
@WindRise29 you can generate the training and validation loss curves (e.g., box_loss, cls_loss, dfl_loss) by loading the `results.csv` from the training run and plotting it using a visualization library such as Matplotlib. Ultralytics YOLO11 doesn't save these plots as images during training by default. However, you can manually extract and visualize the desired metrics post-training. If you'd like automated image saving, you can modify the training script to include this visualization step. Let me know if you need assistance with that!

**Comment by WindRise29** [comment]
> @WindRise29 you can generate the training and validation loss curves (e.g., box_loss, cls_loss, dfl_loss) by loading the `results.csv` from the training run and plotting it using a visualization library such as Matplotlib. Ultralytics YOLO11 doesn't save these plots as images during training by default. However, you can manually extract and visualize the desired metrics post-training. If you'd like automated image saving, you can modify the training script to include this visualization step. Let me know if you need assistance with that!

Thank you very much.

**Comment by Y-T-G** [comment]
It saves automatically after training ends.

**Comment by Y-T-G** [comment]
Or you can use this to plot it using the CSV

https://github.com/ultralytics/ultralytics/blob/9470f91c4d147aa28b6b874e7b439739b1315ace/ultralytics/utils/plotting.py#L1131
